pkgs/by-name/yt/ytui-music/package.nix

```
@@ -0,0 +1,59 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, openssl
+, sqlite
+, stdenv
+, darwin
+, mpv
+, youtube-dl
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "ytui-music-unstable";
+  # Upstream doesn't currently have tags that would be any more meaningful than date of commit
+  # Change this when they upstream has a proper release
+  version = "18-09-2023";
+
+  src = fetchFromGitHub {
+    owner = "sudipghimire533";
+    repo = "ytui-music";
+    rev = version;
+    hash = "sha256-A54xJYgq1j1mfW02pi4KGmuIc38XJOqi9rKvS5EmYb4=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "libmpv-2.0.1" = "sha256-efbXk0oXkzlIqgbP4wKm7sWlVZBT2vzDSN3iwsw2vL0=";
+    };
+  };
+
+  checkFlags = [
+    "--skip=tests::display_config_path"
+    "--skip=tests::inspect_server_list"
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    openssl
+    sqlite
+    mpv
+    youtube-dl
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.CoreFoundation
+    darwin.apple_sdk.frameworks.Security
+  ];
+
```

and add `makeBinaryWrapper` to `nativeBuildInputs`

```suggestion

  postInstall = ''
    wrapProgram $out/bin/ytui_music \
      --prefix PATH : ${lib.makeBinPath [ youtube-dl ]}
  '';

```

---

added `postInstall` in https://github.com/NixOS/nixpkgs/pull/256702/commits/66733f9113ecfd9f63910008c2056bd3767d3676